lib/storage/data/file/DataFileStore.js

```diff
@@ -266,18 +274,30 @@ class DataFileStore {
                   { method: 'get', key, filePath, byteRange });
         const cbOnce = jsutil.once(callback);
         const rs = fs.createReadStream(filePath, readStreamOptions)
-                  .on('error', err => {
-                      if (err.code === 'ENOENT') {
-                          return cbOnce(errors.ObjNotFound);
-                      }
-                      log.error('error retrieving file',
-                          { method: 'get', key, filePath,
-                              error: err });
-                      return cbOnce(
-                          errors.InternalError.customizeDescription(
-                              `filesystem read error: ${err.code}`));
-                  })
-                  .on('open', () => { cbOnce(null, rs); });
+                .on('error', err => {
```

Comment:
Do we need a handler for `close` event where the user abruptly closes the connection? It would be ideal to test the case to see that we are not leaking file handles since `autoClose` is now `false`.

Reply:
+1

Reply:
When the `close` event is triggered, the file descriptor is already closed, or what do you mean exactly?

Also, I've tested with both `autoClose` set to `true` or `false`. In both cases, file descriptors remain open when connections are abruptly closed. However, the file descriptors are eventually closed for both cases (after ~2 minutes) Not sure why though...

This behavior seems to be in the nature of streams => https://github.com/nodejs/node/issues/1834

Reply:
Interesting, there could be a timeout emitted by the underlying socket and the fd is closed after that. Since it's not introduced by this PR I think we should follow up with a change where we close the fd on `close` event. It can be used as an attack vector otherwise.

Reply:
Yeah. Although I was testing with a small script, and the 'close' event doesn't seem to be triggered at all (even with `autoClose: true`) in this scenario. It will probably have to be handled one layer above (http, so probably in the RestServer).